lib/chef/knife/azurerm_base.rb

```diff
@@ -146,18 +174,18 @@ def check_token_validity(token_details)
 
       def validate_azure_login
         err_string = "Please run XPLAT's 'azure login' command OR specify azure_tenant_id, azure_subscription_id, azure_client_id, azure_client_secret in your knife.rb"
-        if Chef::Platform.windows?
+        if Chef::Platform.windows? && (is_old_xplat? || is_WCM_env_var_set?)
```

Comment:
what happens if we're on Windows but we're on a newer version of xplat and `AZURE_USE_SECURE_TOKEN_STORAGE` isn't set?